File: 0290-Word-Pattern/0290.py
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        map_pattern = {}
        map_s = {}
        list_s = s.split()

        if len(pattern) != len(list_s):
            return False
        
        for index, char in enumerate(pattern):
            word = list_s[index]

            if char in map_pattern and map_pattern[char] != word:
                return False

            if word in map_s and map_s[word] != char:
                return False
            
            map_pattern[char] = word
            map_s[word] = char
            
        
        return True

File: 0290-Word-Pattern/0290.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean wordPattern(String pattern, String s) {
        String[] arrayS = s.split(" ");
        if (pattern.length() != arrayS.length)
            return false;

        Map<Character, String> mapPattern = new HashMap<>();
        Map<String, Character> mapS = new HashMap<>();

        for (int i = 0; i < pattern.length(); i++) {
            Character c = pattern.charAt(i);
            String w = arrayS[i];

            if (mapPattern.containsKey(c) && !mapPattern.get(c).equals(w))
                return false;
            if (mapS.containsKey(w) && !mapS.get(w).equals(c))
                return false;

            mapPattern.put(c, w);
            mapS.put(w, c);
        }

        return true;
    }
}